jdaviz/core/template_mixin.py

```diff
@@ -184,6 +184,16 @@ def session(self):
     def data_collection(self):
         return self._app.session.data_collection
 
+    def _clear_cache(self, *attrs):
+        """
+        provide convenience function to clearing the cache for cached_properties
+        """
+        if not len(attrs):
+            attrs = getattr(self, '_cached_properties', [])
```

Comment:
What about something like this? The code can look better. I am just throwing out an idea.

```suggestion
        if not len(attrs):
            attrs = getattr(self, '_cached_properties', [])
        else:
            vetted_attrs = [x for x in attrs if x in getattr(self, '_cached_properties', [])]
            attrs = vetted_attrs
```

Reply:
right, that is essentially what I meant... but then that would mean that you MUST provide `_cached_properties` and anything not in that list cannot be cleared (the original intention of `_cached_properties` was just the ability to set the default "clear all" list).  I think this makes using this more complicated, but we can do it if necessary (in which case I would definitely want to put this somewhere more general if the logic is getting more advanced).  Can you show me a case where not having this check would be a problem?

Reply:
Well, theoretically any of these could be deleted by accident, right?

```python
>>> plg = imviz.plugins["Compass"]
>>> sorted(plg._obj.__dict__)
['_app',
 '_cross_validation_lock',
 '_model_id',
 '_plugin_name',
 '_template_metadata',
 '_trait_notifiers',
 '_trait_validators',
 '_trait_values',
 '_viewer_callbacks',
 'bqplot_figs_resize',
 'viewer']
```

Reply:
right, but the user never calls this.  Even with the check, they could "accidentally" be added to `_cached_properties` and still deleted.  Or we could accidentally call `del` on them outside this method.  I'll take a look at the inspection option and see if the overhead is negligible.